crates/sui-node/src/metrics.rs

```diff
@@ -119,21 +124,28 @@ pub fn start_metrics_push_task(config: &sui_config::NodeConfig, registry: Regist
         url: &reqwest::Url,
         registry: &RegistryService,
     ) -> Result<(), anyhow::Error> {
-        let metrics = TextEncoder
-            .encode_to_string(&registry.gather_all())
-            .context("encoding metrics")?;
+        let mut buf: Vec<u8> = vec![];
+        let encoder = prometheus::ProtobufEncoder::new();
+        encoder.encode(&registry.gather_all(), &mut buf)?;
 
         let response = client
             .client()
             .post(url.to_owned())
-            .body(metrics)
+            .header(header::CONTENT_TYPE, "application/mysten.proxy")
```

Comment:
You sure you don't want to use the proper content type here? https://docs.rs/prometheus/latest/prometheus/constant.PROTOBUF_FORMAT.html

Reply:
sure. no problem. it only matters on the ingest side that this posts to.  the proxy looks for the above, but we can use the constant.